tls/s2n_tls13_secrets.c

```diff
@@ -671,3 +702,51 @@ S2N_RESULT s2n_tls13_secrets_get(struct s2n_connection *conn, s2n_extract_secret
     RESULT_ENSURE_GT(secret->size, 0);
     return S2N_RESULT_OK;
 }
+
+int s2n_connection_tls_exporter(struct s2n_connection *conn,
+        const uint8_t *label_in, uint32_t label_length,
+        const uint8_t *context, uint32_t context_length,
+        uint8_t *output_in, uint32_t output_length)
+{
+    POSIX_ENSURE_REF(conn);
+    POSIX_ENSURE_REF(output_in);
+    POSIX_ENSURE_REF(label_in);
+    POSIX_ENSURE_REF(context);
+    POSIX_ENSURE(s2n_connection_get_protocol_version(conn) == S2N_TLS13, S2N_ERR_INVALID_STATE);
+
+    POSIX_ENSURE(is_handshake_complete(conn), S2N_ERR_HANDSHAKE_NOT_COMPLETE);
+    POSIX_ENSURE_REF(conn->secure);
+    POSIX_ENSURE_REF(conn->secure->cipher_suite);
+    s2n_hmac_algorithm hmac_alg = conn->secure->cipher_suite->prf_alg;
+
+    struct s2n_blob label = { 0 };
+    POSIX_GUARD(s2n_blob_init(&label, (uint8_t *) label_in, label_length));
+
+    uint8_t derived_secret_bytes[S2N_TLS13_SECRET_MAX_LEN] = { 0 };
+    struct s2n_blob derived_secret = { 0 };
+    POSIX_GUARD(s2n_blob_init(&derived_secret, derived_secret_bytes,
+            s2n_get_hash_len(CONN_HMAC_ALG(conn))));
```

Comment:
Nit, but I would suggest verifying that sizeof(derived_secret_bytes) <= s2n_get_hash_len(CONN_HMAC_ALG(conn)) before you use the latter as the definitive size of derived_secret_bytes.